livekit/src/rtc_engine/mod.rs

```diff
@@ -402,15 +404,25 @@ impl EngineInner {
     async fn on_session_event(self: &Arc<Self>, event: SessionEvent) -> EngineResult<()> {
         match event {
             SessionEvent::Close { source, reason, action, retry_now } => {
-                log::warn!("received session close: {:?} {:?} {:?}", source, reason, action);
                 match action {
-                    proto::leave_request::Action::Resume => {
-                        self.reconnection_needed(retry_now, false)
-                    }
-                    proto::leave_request::Action::Reconnect => {
-                        self.reconnection_needed(retry_now, true)
+                    proto::leave_request::Action::Resume
+                    | proto::leave_request::Action::Reconnect => {
+                        log::warn!(
+                            "received session close: {:?} {:?} {:?}",
```

Comment:
```suggestion
                            "received server-initiated reconnect: {:?} {:?} {:?}",
```

Reply:
it isn't necessarily server-initiated, this part needs cleanup but it is also called when the SignalClient is being closed unexpectedly. In this case we create our own leave_request locally with the reason Unknown.